src/science.jl

```diff
@@ -13,24 +13,27 @@ function ser{N}(x::Array{Float64,N}, thresh::Float64=0.01)
   reshape(SER, dims[2:end])
 end
 
-function r1eff(S::Array{Float64,3}, R10::Matrix{Float64}, TR::Float64, flip::Float64)
+function r1eff{M,N}(S::Array{Float64,M}, R10::Array{Float64,N}, TR::Float64, flip::Float64)
   @dprint "converting DCE signal to effective R1"
   @assert 0.0 < flip "flip angle must be positive"
   @assert 0.0 < TR && TR < 1.0 "TR must be in units of ms"
-  S0 = squeeze(mean(S[1:2,:,:],1),1)
-  nt, nx, ny = size(S)
+  dims = size(S)
+  nt = dims[1]
+  n = prod(dims[2:end])
+  S = reshape(S, (nt, n))
+  S0 = mean(S[1:2,:],1)
```

Comment:
Good idea. But let's wait until this has been merged and keep this PR topical.  I will test the validation routines on your fork to check that no results have been changed.